src/test/java/com/jcabi/log/VerboseProcessTest.java

```diff
@@ -310,6 +313,81 @@ public void terminatesMonitorsAndProcessIfClosedNormal() throws Exception {
     }
 
     /**
+     * VerboseProcess can run a java process which will throw a stack trace and
+     * makes sure it will group the stack trace.
+     */
+    @Test
+    public void logCompleteStackTrace() {
+        final org.apache.log4j.Logger logger =
+                org.apache.log4j.Logger.getRootLogger();
+        final TestAppender appender = new TestAppender();
+        logger.addAppender(appender);
+        final String[] commands = {
+            retrieveJavaExecLocation(), "-cp",
+            System.getProperty("java.class.path"),
+            "com.jcabi.log.VerboseProcessExample",
+        };
+        final ProcessBuilder builder = new ProcessBuilder(commands);
+        final VerboseProcess process = new VerboseProcess(
+            builder, Level.INFO, Level.SEVERE
+        );
+        boolean failed = false;
+        try {
+            process.stdout();
+        } catch (final IllegalArgumentException ex) {
+            failed = true;
+            MatcherAssert.assertThat(
+                    ex.getMessage(),
+                    Matchers.containsString(VerboseProcessExample.SYSOUT_1)
+            );
+            MatcherAssert.assertThat(
+                    ex.getMessage(),
+                    Matchers.containsString(VerboseProcessExample.SYSOUT_2)
+            );
+        } finally {
+            logger.removeAppender(appender);
+            process.close();
+        }
+        Assert.assertTrue("Process should have failed!", failed);
+        verifyLogs(appender);
+    }
+
+    /**
+     * Gets the location of Java, whether on Linux of Windows.
+     * @return String with Java location
+     */
+    private static String retrieveJavaExecLocation() {
+        final String rootpath = System.getProperty("java.home");
+        final String finalpath;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            finalpath = String.format("%s%s", rootpath, "\\bin\\java.exe");
+        } else {
+            finalpath = String.format("%s%s", rootpath, "/bin/java");
+        }
+        final File file = new File(finalpath);
+        if (file.exists()) {
+            return finalpath;
+        }
+        throw new IllegalStateException("Unable to get the Java Path.");
+    }
+
+    /**
+     * Checks appender to make sure expected log statements are present.
+     * @param appender Log appender
+     */
+    private static void verifyLogs(final TestAppender appender) {
```

Comment:
@dean-e-clark `TestAppender` should be qualified here, see https://github.com/teamed/qulice/issues/738